### ISSUE TITLE
Egress gateway example from docs not working on GKE

### ISSUE THREAD

**Issue by MarcoSchaefer** [istio/istio#25099]
**Bug description**
After creating a new cluster on GKE and following the steps in https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/, requesting `http://edition.cnn.com/politics` (the URL from the example) returns a `503` instead of `200`.

Cluster's Istio version: `1.2.10`.
Local Istio client version: `1.6.3`.

**Affected product area (please put an X in all that apply)**

[X] Configuration Infrastructure
[X] Docs
[X] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**
`curl -sL -o /dev/null -D - http://edition.cnn.com/politics` should return a status code 200.

**Steps to reproduce the bug**
- Create a new cluster on GKE.
- Follow the steps described in https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
istioctl version --remote
client version: 1.6.3
citadel version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
galley version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
pilot version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
policy version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
sidecar-injector version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
telemetry version: c834c448a92e691d71b42a966d94fb1822ac10e1-dirty
pilot version: 1.6.3
data plane version: 1.0-dev (1 proxies), 1.6.3 (2 proxies)
```
However, the images provided by GKE are `1.2.10`.

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"archive", BuildDate:"2020-02-29T16:37:45Z", GoVersion:"go1.14", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-gke.36", GitCommit:"34a615f32e9a0c9e97cdb9f749adb392758349a6", GitTreeState:"clean", BuildDate:"2020-04-06T16:33:17Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
Istio was provided by GKE itself (by selecting the checkbox `Istio` while creating the cluster on GKE).

**Environment where bug was observed (cloud vendor, OS, etc)**
GCP

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.
[istio-dump.tar.gz](https://github.com/istio/istio/files/4848287/istio-dump.tar.gz)

**Comment by MarcoSchaefer** [comment]
Deploying the gateway using the helm templates from the same version as the cluster (1.2.10) made everything work like a charm! :smiley:

**Comment by kpiyush17** [comment]
@MarcoSchaefer I am facing the same issue of 503 in AWS EKS. Have you tried the example on EKS?

**Comment by MarcoSchaefer** [comment]
Hi @kpiyush17, I tried it only on GKE. In the end, the issue was caused by using a wrong version of the egress gateway. I noticed that I was receiving a 503 when the egress gateway couldn't be reached through it's service (when it's pod was down or when the version was a wrong one).
Instead of following the example described in https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/, I should have used the one described in https://istio.io/v1.2/docs/tasks/traffic-management/egress/egress-gateway/.

I hope this helps.
